src/AutoRest.CSharp/Generation/Writers/ClientWriter.cs

```diff
@@ -134,12 +200,71 @@ private void WriteClientFields(CodeWriter writer, Client client)
+                using (writer.Scope())
+                {
+                    writer.Line($"{typeof(Argument)}.AssertNotNull({EndpointVariable}, nameof({EndpointVariable}));");
+                    writer.Line($"{typeof(Argument)}.AssertNotNull({CredentialVariable}, nameof({CredentialVariable}));");
+                    writer.Line();
+
+                    writer.Line($"{OptionsVariable} ??= new {client.Type.Name}Options();");
+                    writer.Line($"{ClientDiagnosticsField} = new {typeof(ClientDiagnostics)}({OptionsVariable});");
+                    writer.Line($"{PipelineField} = {typeof(HttpPipelineBuilder)}.Build({OptionsVariable}, new {typeof(AzureKeyCredentialPolicy)}({CredentialVariable}, \"api-key\"));");
+                    writer.Append($"this.RestClient = new {client.RestClient.Type}({ClientDiagnosticsField}, {PipelineField}, {EndpointVariable});");
+                }
+                writer.Line();
+            }
+
+            if (configuration.CredentialTypes.Contains("TokenCredential", StringComparer.OrdinalIgnoreCase))
+            {
+                writer.WriteXmlDocumentationSummary($"Initializes a new instance of {client.Type.Name}");
+                writer.WriteXmlDocumentationParameter(EndpointVariable, "server parameter.");
+                writer.WriteXmlDocumentationParameter(CredentialVariable, "A credential used to authenticate to an Azure Service.");
+                writer.WriteXmlDocumentationParameter(OptionsVariable, "The options for configuring the client.");
+                writer.Append($"public {client.Type.Name:D}({typeof(Uri)} {EndpointVariable}, {typeof(TokenCredential)} {CredentialVariable}, {client.Type.Name}Options {OptionsVariable} = null)");
+                using (writer.Scope())
+                {
+                    writer.Line($"{typeof(Argument)}.AssertNotNull({EndpointVariable}, nameof({EndpointVariable}));");
+                    writer.Line($"{typeof(Argument)}.AssertNotNull({CredentialVariable}, nameof({CredentialVariable}));");
+                    writer.Line();
+
+                    writer.Line($"{OptionsVariable} ??= new {client.Type.Name}Options();");
+                    writer.Line($"{ClientDiagnosticsField} = new {typeof(ClientDiagnostics)}({OptionsVariable});");
+                    var scopesParam = new CodeWriterDeclaration("scopes");
+                    string scope = "";
+                    if (configuration.CredentialScopes != null)
+                    {
+                        if (configuration.CredentialScopes.Length == 1)
+                        {
+                            scope = $"\"{configuration.CredentialScopes[0]}\"";
+                        }
+                        else
+                        {
+                            string scopesArr = "{\"" + string.Join("\", \"", configuration.CredentialScopes) + "\"}";
```

Comment:
Don't concatenate strings, use the writer with the literal flag to get escaping.